okhttp-tests/src/test/java/okhttp3/internal/framed/Http2Test.java

```diff
@@ -277,10 +277,13 @@ public void pushPromise(int streamId, int promisedStreamId, List<Header> headerB
     frame.writeInt(1);
 
     try {
-      fr.nextFrame(new BaseTestHandler());
-      fail();
+      fr.nextFrame(new BaseTestHandler() {
+        @Override public void settings(boolean clearPrevious, Settings settings) {
+          assert settings.get(7) == 1;
```

Comment:
I would look at other tests to see how they do this. I personally would use an AtomicInteger outside the class, set its value in the callback here, and assert its value below (using `assertEquals()`, not `assert`).